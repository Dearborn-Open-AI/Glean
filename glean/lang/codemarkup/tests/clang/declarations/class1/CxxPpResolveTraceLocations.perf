{
  "@generated": null,
  "cxx1.DeclarationNameSpan.5": 11,
  "cxx1.RecordDefinition.5": 8,
  "cxx1.Trace.5": 2
}