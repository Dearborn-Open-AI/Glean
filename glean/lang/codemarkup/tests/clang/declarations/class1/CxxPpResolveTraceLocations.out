[
  "@generated",
  {
    "key": {
      "tuplefield0": {
        "name": "",
        "file": { "key": "test.cpp" },
        "location": {
          "range": {
            "file": { "key": "test.cpp" },
            "lineBegin": 12,
            "columnBegin": 5,
            "lineEnd": 12,
            "columnEnd": 12
          }
        },
        "span": { "start": 264, "length": 1 }
      },
      "tuplefield1": {
        "cxx": {
          "decl": {
            "variable": {
              "key": {
                "name": {
                  "key": {
                    "name": { "key": "" },
                    "scope": {
                      "local": {
                        "key": {
                          "name": { "key": { "constructor": { } } },
                          "scope": {
                            "recordWithAccess": {
                              "record": {
                                "key": {
                                  "name": { "key": "U" },
                                  "scope": { "global_": { } }
                                }
                              },
                              "access": 0
                            }
                          }
                        }
                      }
                    }
                  }
                },
                "type": { "key": "const U &" },
                "kind": { "local": { "kind": 2, "attribute": 0 } },
                "source": {
                  "file": { "key": "test.cpp" },
                  "lineBegin": 12,
                  "columnBegin": 5,
                  "lineEnd": 12,
                  "columnEnd": 12
                }
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "tuplefield0": {
        "name": "",
        "file": { "key": "test.cpp" },
        "location": {
          "range": {
            "file": { "key": "test.cpp" },
            "lineBegin": 13,
            "columnBegin": 16,
            "lineEnd": 13,
            "columnEnd": 23
          }
        },
        "span": { "start": 290, "length": 1 }
      },
      "tuplefield1": {
        "cxx": {
          "decl": {
            "variable": {
              "key": {
                "name": {
                  "key": {
                    "name": { "key": "" },
                    "scope": {
                      "local": {
                        "key": {
                          "name": { "key": { "operator_": "operator=" } },
                          "scope": {
                            "recordWithAccess": {
                              "record": {
                                "key": {
                                  "name": { "key": "U" },
                                  "scope": { "global_": { } }
                                }
                              },
                              "access": 0
                            }
                          }
                        }
                      }
                    }
                  }
                },
                "type": { "key": "const U &" },
                "kind": { "local": { "kind": 2, "attribute": 0 } },
                "source": {
                  "file": { "key": "test.cpp" },
                  "lineBegin": 13,
                  "columnBegin": 16,
                  "lineEnd": 13,
                  "columnEnd": 23
                }
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "tuplefield0": {
        "name": "A",
        "file": { "key": "test.cpp" },
        "location": {
          "range": {
            "file": { "key": "test.cpp" },
            "lineBegin": 16,
            "columnBegin": 1,
            "lineEnd": 18,
            "columnEnd": 1
          }
        },
        "span": { "start": 304, "length": 1 }
      },
      "tuplefield1": {
        "cxx": {
          "defn": {
            "record_": {
              "key": {
                "declaration": {
                  "key": {
                    "name": {
                      "key": {
                        "name": { "key": "A" },
                        "scope": { "global_": { } }
                      }
                    },
                    "kind": { "struct_": { } },
                    "source": {
                      "file": { "key": "test.cpp" },
                      "lineBegin": 16,
                      "columnBegin": 1,
                      "lineEnd": 18,
                      "columnEnd": 1
                    }
                  }
                },
                "bases": [ ],
                "members": {
                  "key": [
                    {
                      "function_": {
                        "key": {
                          "name": {
                            "key": {
                              "name": { "key": { "destructor": { } } },
                              "scope": {
                                "recordWithAccess": {
                                  "record": {
                                    "key": {
                                      "name": { "key": "A" },
                                      "scope": { "global_": { } }
                                    }
                                  },
                                  "access": 0
                                }
                              }
                            }
                          },
                          "signature": {
                            "key": {
                              "returns": { "key": "void" },
                              "parameters": [ ]
                            }
                          },
                          "method": {
                            "isVirtual": true,
                            "isConst": false,
                            "isVolatile": false,
                            "refQualifier": 0
                          },
                          "source": {
                            "file": { "key": "test.cpp" },
                            "lineBegin": 17,
                            "columnBegin": 3,
                            "lineEnd": 17,
                            "columnEnd": 18
                          }
                        }
                      }
                    }
                  ]
                }
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "tuplefield0": {
        "name": "B",
        "file": { "key": "test.cpp" },
        "location": {
          "range": {
            "file": { "key": "test.cpp" },
            "lineBegin": 20,
            "columnBegin": 1,
            "lineEnd": 22,
            "columnEnd": 1
          }
        },
        "span": { "start": 339, "length": 1 }
      },
      "tuplefield1": {
        "cxx": {
          "defn": {
            "record_": {
              "key": {
                "declaration": {
                  "key": {
                    "name": {
                      "key": {
                        "name": { "key": "B" },
                        "scope": { "global_": { } }
                      }
                    },
                    "kind": { "struct_": { } },
                    "source": {
                      "file": { "key": "test.cpp" },
                      "lineBegin": 20,
                      "columnBegin": 1,
                      "lineEnd": 22,
                      "columnEnd": 1
                    }
                  }
                },
                "bases": [
                  {
                    "base": {
                      "key": {
                        "name": {
                          "key": {
                            "name": { "key": "A" },
                            "scope": { "global_": { } }
                          }
                        },
                        "kind": { "struct_": { } },
                        "source": {
                          "file": { "key": "test.cpp" },
                          "lineBegin": 16,
                          "columnBegin": 1,
                          "lineEnd": 18,
                          "columnEnd": 1
                        }
                      }
                    },
                    "access": 0,
                    "isVirtual": false
                  }
                ],
                "members": {
                  "key": [
                    {
                      "variable": {
                        "key": {
                          "name": {
                            "key": {
                              "name": { "key": "u" },
                              "scope": {
                                "recordWithAccess": {
                                  "record": {
                                    "key": {
                                      "name": { "key": "B" },
                                      "scope": { "global_": { } }
                                    }
                                  },
                                  "access": 0
                                }
                              }
                            }
                          },
                          "type": { "key": "U" },
                          "kind": { "field": { "mutable_": false } },
                          "source": {
                            "file": { "key": "test.cpp" },
                            "lineBegin": 21,
                            "columnBegin": 3,
                            "lineEnd": 21,
                            "columnEnd": 5
                          }
                        }
                      }
                    }
                  ]
                }
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "tuplefield0": {
        "name": "T",
        "file": { "key": "test.cpp" },
        "location": {
          "range": {
            "file": { "key": "test.cpp" },
            "lineBegin": 9,
            "columnBegin": 1,
            "lineEnd": 9,
            "columnEnd": 11
          }
        },
        "span": { "start": 228, "length": 1 }
      },
      "tuplefield1": {
        "cxx": {
          "defn": {
            "record_": {
              "key": {
                "declaration": {
                  "key": {
                    "name": {
                      "key": {
                        "name": { "key": "T" },
                        "scope": { "global_": { } }
                      }
                    },
                    "kind": { "struct_": { } },
                    "source": {
                      "file": { "key": "test.cpp" },
                      "lineBegin": 9,
                      "columnBegin": 1,
                      "lineEnd": 9,
                      "columnEnd": 11
                    }
                  }
                },
                "bases": [ ],
                "members": { "key": [ ] }
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "tuplefield0": {
        "name": "U",
        "file": { "key": "test.cpp" },
        "location": {
          "range": {
            "file": { "key": "test.cpp" },
            "lineBegin": 10,
            "columnBegin": 1,
            "lineEnd": 14,
            "columnEnd": 1
          }
        },
        "span": { "start": 241, "length": 1 }
      },
      "tuplefield1": {
        "cxx": {
          "defn": {
            "record_": {
              "key": {
                "declaration": {
                  "key": {
                    "name": {
                      "key": {
                        "name": { "key": "U" },
                        "scope": { "global_": { } }
                      }
                    },
                    "kind": { "struct_": { } },
                    "source": {
                      "file": { "key": "test.cpp" },
                      "lineBegin": 10,
                      "columnBegin": 1,
                      "lineEnd": 14,
                      "columnEnd": 1
                    }
                  }
                },
                "bases": [ ],
                "members": {
                  "key": [
                    {
                      "function_": {
                        "key": {
                          "name": {
                            "key": {
                              "name": { "key": { "constructor": { } } },
                              "scope": {
                                "recordWithAccess": {
                                  "record": {
                                    "key": {
                                      "name": { "key": "U" },
                                      "scope": { "global_": { } }
                                    }
                                  },
                                  "access": 0
                                }
                              }
                            }
                          },
                          "signature": {
                            "key": {
                              "returns": { "key": "void" },
                              "parameters": [ ]
                            }
                          },
                          "method": {
                            "isVirtual": false,
                            "isConst": false,
                            "isVolatile": false,
                            "refQualifier": 0
                          },
                          "source": {
                            "file": { "key": "test.cpp" },
                            "lineBegin": 11,
                            "columnBegin": 3,
                            "lineEnd": 11,
                            "columnEnd": 5
                          }
                        }
                      }
                    },
                    {
                      "function_": {
                        "key": {
                          "name": {
                            "key": {
                              "name": { "key": { "constructor": { } } },
                              "scope": {
                                "recordWithAccess": {
                                  "record": {
                                    "key": {
                                      "name": { "key": "U" },
                                      "scope": { "global_": { } }
                                    }
                                  },
                                  "access": 0
                                }
                              }
                            }
                          },
                          "signature": {
                            "key": {
                              "returns": { "key": "void" },
                              "parameters": [
                                {
                                  "name": { "key": "" },
                                  "type": { "key": "const U &" }
                                }
                              ]
                            }
                          },
                          "method": {
                            "isVirtual": false,
                            "isConst": false,
                            "isVolatile": false,
                            "refQualifier": 0
                          },
                          "source": {
                            "file": { "key": "test.cpp" },
                            "lineBegin": 12,
                            "columnBegin": 3,
                            "lineEnd": 12,
                            "columnEnd": 13
                          }
                        }
                      }
                    },
                    {
                      "function_": {
                        "key": {
                          "name": {
                            "key": {
                              "name": { "key": { "operator_": "operator=" } },
                              "scope": {
                                "recordWithAccess": {
                                  "record": {
                                    "key": {
                                      "name": { "key": "U" },
                                      "scope": { "global_": { } }
                                    }
                                  },
                                  "access": 0
                                }
                              }
                            }
                          },
                          "signature": {
                            "key": {
                              "returns": { "key": "U &" },
                              "parameters": [
                                {
                                  "name": { "key": "" },
                                  "type": { "key": "const U &" }
                                }
                              ]
                            }
                          },
                          "method": {
                            "isVirtual": false,
                            "isConst": false,
                            "isVolatile": false,
                            "refQualifier": 0
                          },
                          "source": {
                            "file": { "key": "test.cpp" },
                            "lineBegin": 13,
                            "columnBegin": 3,
                            "lineEnd": 13,
                            "columnEnd": 24
                          }
                        }
                      }
                    }
                  ]
                }
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "tuplefield0": {
        "name": "constructor",
        "file": { "key": "test.cpp" },
        "location": {
          "range": {
            "file": { "key": "test.cpp" },
            "lineBegin": 11,
            "columnBegin": 3,
            "lineEnd": 11,
            "columnEnd": 5
          }
        },
        "span": { "start": 247, "length": 1 }
      },
      "tuplefield1": {
        "cxx": {
          "decl": {
            "function_": {
              "key": {
                "name": {
                  "key": {
                    "name": { "key": { "constructor": { } } },
                    "scope": {
                      "recordWithAccess": {
                        "record": {
                          "key": {
                            "name": { "key": "U" },
                            "scope": { "global_": { } }
                          }
                        },
                        "access": 0
                      }
                    }
                  }
                },
                "signature": {
                  "key": { "returns": { "key": "void" }, "parameters": [ ] }
                },
                "method": {
                  "isVirtual": false,
                  "isConst": false,
                  "isVolatile": false,
                  "refQualifier": 0
                },
                "source": {
                  "file": { "key": "test.cpp" },
                  "lineBegin": 11,
                  "columnBegin": 3,
                  "lineEnd": 11,
                  "columnEnd": 5
                }
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "tuplefield0": {
        "name": "constructor",
        "file": { "key": "test.cpp" },
        "location": {
          "range": {
            "file": { "key": "test.cpp" },
            "lineBegin": 12,
            "columnBegin": 3,
            "lineEnd": 12,
            "columnEnd": 13
          }
        },
        "span": { "start": 254, "length": 1 }
      },
      "tuplefield1": {
        "cxx": {
          "decl": {
            "function_": {
              "key": {
                "name": {
                  "key": {
                    "name": { "key": { "constructor": { } } },
                    "scope": {
                      "recordWithAccess": {
                        "record": {
                          "key": {
                            "name": { "key": "U" },
                            "scope": { "global_": { } }
                          }
                        },
                        "access": 0
                      }
                    }
                  }
                },
                "signature": {
                  "key": {
                    "returns": { "key": "void" },
                    "parameters": [
                      { "name": { "key": "" }, "type": { "key": "const U &" } }
                    ]
                  }
                },
                "method": {
                  "isVirtual": false,
                  "isConst": false,
                  "isVolatile": false,
                  "refQualifier": 0
                },
                "source": {
                  "file": { "key": "test.cpp" },
                  "lineBegin": 12,
                  "columnBegin": 3,
                  "lineEnd": 12,
                  "columnEnd": 13
                }
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "tuplefield0": {
        "name": "destructor",
        "file": { "key": "test.cpp" },
        "location": {
          "range": {
            "file": { "key": "test.cpp" },
            "lineBegin": 17,
            "columnBegin": 3,
            "lineEnd": 17,
            "columnEnd": 18
          }
        },
        "span": { "start": 318, "length": 1 }
      },
      "tuplefield1": {
        "cxx": {
          "decl": {
            "function_": {
              "key": {
                "name": {
                  "key": {
                    "name": { "key": { "destructor": { } } },
                    "scope": {
                      "recordWithAccess": {
                        "record": {
                          "key": {
                            "name": { "key": "A" },
                            "scope": { "global_": { } }
                          }
                        },
                        "access": 0
                      }
                    }
                  }
                },
                "signature": {
                  "key": { "returns": { "key": "void" }, "parameters": [ ] }
                },
                "method": {
                  "isVirtual": true,
                  "isConst": false,
                  "isVolatile": false,
                  "refQualifier": 0
                },
                "source": {
                  "file": { "key": "test.cpp" },
                  "lineBegin": 17,
                  "columnBegin": 3,
                  "lineEnd": 17,
                  "columnEnd": 18
                }
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "tuplefield0": {
        "name": "operator=",
        "file": { "key": "test.cpp" },
        "location": {
          "range": {
            "file": { "key": "test.cpp" },
            "lineBegin": 13,
            "columnBegin": 3,
            "lineEnd": 13,
            "columnEnd": 24
          }
        },
        "span": { "start": 272, "length": 8 }
      },
      "tuplefield1": {
        "cxx": {
          "decl": {
            "function_": {
              "key": {
                "name": {
                  "key": {
                    "name": { "key": { "operator_": "operator=" } },
                    "scope": {
                      "recordWithAccess": {
                        "record": {
                          "key": {
                            "name": { "key": "U" },
                            "scope": { "global_": { } }
                          }
                        },
                        "access": 0
                      }
                    }
                  }
                },
                "signature": {
                  "key": {
                    "returns": { "key": "U &" },
                    "parameters": [
                      { "name": { "key": "" }, "type": { "key": "const U &" } }
                    ]
                  }
                },
                "method": {
                  "isVirtual": false,
                  "isConst": false,
                  "isVolatile": false,
                  "refQualifier": 0
                },
                "source": {
                  "file": { "key": "test.cpp" },
                  "lineBegin": 13,
                  "columnBegin": 3,
                  "lineEnd": 13,
                  "columnEnd": 24
                }
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "tuplefield0": {
        "name": "u",
        "file": { "key": "test.cpp" },
        "location": {
          "range": {
            "file": { "key": "test.cpp" },
            "lineBegin": 21,
            "columnBegin": 3,
            "lineEnd": 21,
            "columnEnd": 5
          }
        },
        "span": { "start": 351, "length": 1 }
      },
      "tuplefield1": {
        "cxx": {
          "decl": {
            "variable": {
              "key": {
                "name": {
                  "key": {
                    "name": { "key": "u" },
                    "scope": {
                      "recordWithAccess": {
                        "record": {
                          "key": {
                            "name": { "key": "B" },
                            "scope": { "global_": { } }
                          }
                        },
                        "access": 0
                      }
                    }
                  }
                },
                "type": { "key": "U" },
                "kind": { "field": { "mutable_": false } },
                "source": {
                  "file": { "key": "test.cpp" },
                  "lineBegin": 21,
                  "columnBegin": 3,
                  "lineEnd": 21,
                  "columnEnd": 5
                }
              }
            }
          }
        }
      }
    }
  }
]